src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -173,22 +196,25 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
 
         public async Task ConnectAsync(IConnectionViewModel connection) {
             _shell.AssertIsOnMainThread();
-            await _connectionManager.ConnectAsync(connection);
+            var progressBarMessage = Resources.ConnectionManager_SwitchConnectionProgressBarMessage.FormatInvariant(_connectionManager.ActiveConnection.Name, connection.Name);
+            using (var progressBarSession = _shell.ShowProgressBar(progressBarMessage)) {
```

Comment:
It isn't really flickering for local on my machine, cause even local requires 2 processes to start (broker and host).
But I can add a delay. How big it should be?

---
In reply to: [79271460](https://github.com/Microsoft/RTVS/pull/2342#discussion_r79271460) [](ancestors = 79271460)